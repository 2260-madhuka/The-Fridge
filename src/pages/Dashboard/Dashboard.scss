$font-stack: Helvetica, sans-serif;
$primary-color_1: #ffffff;
$primary-color_2: #f4f9ff;

.screen {
    background-color: $primary-color_2;
    display: flex;
    flex-direction: row;
    justify-content: center;
    width: 100%;

    .div {
        background-color: $primary-color_2;
        border: none;
        height: 900px;
        position: relative;
        width: 1440px;
    }

    .overlap {
        height: 444px;
        left: 0;
        position: absolute;
        top: 0;
        width: 1440px;
    }

    .rectangle {
        background-color: $primary-color_1;
        height: 353px;
        left: 0;
        position: absolute;
        top: 0;
        width: 1440px;
    }

    .form-subText {
        color: #728197;
        font-size: 12px;
        font-family: Inter;
        font-style: normal;
        font-weight: 500;
        line-height: normal;
        text-transform: capitalize;
        margin-left: 65px;
    }

    .form {
        height: 170px;
        left: 229px;
        position: absolute;
        top: 220px;
        width: 982px;
        display: flex;
        flex-direction: column;
        justify-content: center;
        background-color: $primary-color_1;
        box-shadow: 10px -4px 60px #0000001f;

        &-subText-container {
            margin-top: 20px;
        }

        &-header {
            display: flex;
            flex-direction: row;
            justify-content: center;
            align-items: center;
        }
    }

    .subheading {
        color: #728197;
        font-family: "Inter-Medium", Helvetica;
        font-size: 18px;
        font-weight: 500;
        left: 531px;
        letter-spacing: -0.54px;
        line-height: normal;
        position: absolute;
        text-align: center;
        top: 145px;
    }

    .heading {
        color: #003a59;
        font-family: "Inter-ExtraBold", Helvetica;
        font-size: 36px;
        font-weight: 800;
        left: 532px;
        letter-spacing: -1.08px;
        line-height: normal;
        position: absolute;
        text-align: center;
        top: 89px;
    }

    .text-wrapper {
        color: #2e3849;
        font-family: "Inter-SemiBold", Helvetica;
        font-size: 14px;
        font-weight: 600;
        left: 1079px;
        letter-spacing: -0.42px;
        line-height: normal;
        position: absolute;
        text-align: right;
        top: 427px;
    }

    .item {
        background-color: $primary-color_1;
        border: 1px solid;
        border-color: #eef3f8;
        border-radius: 4px;
        box-shadow: 0px 6px 24px #0048730a;
        height: 50px;
        left: 253px;
        position: relative;
        top: 461px;
        width: 934px;
        margin-bottom: 8px;
    }

    .expiry-date {
        color: #728197;
        font-family: "Inter-SemiBold", Helvetica;
        font-size: 12px;
        font-weight: 600;
        left: 157px;
        letter-spacing: -0.36px;
        line-height: normal;
        position: absolute;
        top: 16px;
    }

    .delete-outline-black {
        height: 24px;
        left: 889px;
        position: absolute;
        top: 12px;
        width: 24px;
    }

    .title {
        color: #2e3849;
        font-family: "Inter-SemiBold", Helvetica;
        font-size: 16px;
        font-weight: 600;
        left: 19px;
        letter-spacing: -0.48px;
        line-height: normal;
        position: absolute;
        top: 14px;
        white-space: nowrap;
    }

    .chip {
        height: 24px;
        left: 777px;
        position: absolute;
        top: 12px;
        width: 102px;
    }

    .overlap-group-healthy {
        background-color: #DBFFE6;
        border-radius: 22px;
        height: 24px;
        position: relative;
        width: 100px;
    }

    .overlap-group-expired {
        background-color: #FFEBEB;
        border-radius: 22px;
        height: 24px;
        position: relative;
        width: 100px;
    }

    .overlap-group-expiring-soon {
        background-color: #fffdcc;
        border-radius: 22px;
        height: 24px;
        position: relative;
        width: 100px;
    }

    .text {
        font-family: "Inter-SemiBold", Helvetica;
        font-size: 12px;
        font-weight: 600;
        left: 13px;
        letter-spacing: -0.36px;
        line-height: normal;
        position: absolute;
        text-align: center;
        top: 5px;
    }

    .text-expiring-soon {
        color: #754311;
    }

    .text-healthy {
        color: #23553E;
    }

    .text-expired {
        color: #752B2B;
    }

    .loding {
        color: #003A59;
        font-family: "Inter-SemiBold", Helvetica;
        font-size: 14px;
        font-weight: 600;
        // left: 1079px;
        letter-spacing: -0.42px;
        line-height: normal;
        position: relative;
        text-align: right;

        display: flex;
        justify-content: center;
        align-items: center;

        top: 556px;
    }
}