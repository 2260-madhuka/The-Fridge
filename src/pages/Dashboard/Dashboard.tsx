import React, { useEffect, useState } from "react";
import { TheFridgeTextInput, TheFridgeButton } from "../../components";
import { DeleteIcon } from "../../assets/icons";
import "./Dashboard.scss";
import axios from 'axios'
import { URL } from '../../models/enums'
import { expiryDateCheck } from '../../utils/helpers/helpers'

export interface DashboardProps { }

const Dashboard: React.FC<DashboardProps> = () => {

    const [apiData, setApiData] = useState([] as any[]);
    const [itemName, setItemName] = useState<string>('');
    const [expiryDate, setExpiryDate] = useState<string>('');
    const [isLoading, setIsLoading] = useState<boolean>(true);

    useEffect(() => {
        getAllDataFromAPI();
    }, []);

    // Get all data from API
    const getAllDataFromAPI = async () => {
        try {
            const response = await axios.get(URL.BASE_URL);
            const data = response.data;
            setIsLoading(false);
            setApiData(data)
        } catch (error) {
            console.error(error);
        }
    };
    // Get total data from API
    const dataLength = apiData.length;

    // Post data to API
    const postItemToAPI = async () => {
        try {
            const newItem = {
                title: itemName,
                expiry: expiryDate
            };
            const response = await axios.post(URL.BASE_URL, newItem);
            const responseData = response.data;
            setApiData([...apiData, responseData]);
        } catch (error) {
            console.error(error);
        }
    };

    const handleAddToFridge = () => {
        postItemToAPI();
        setItemName('');
        setExpiryDate('');
    };

    return (
        <div className='screen'>
            <div className='div'>
                <div className='overlap'>
                    <div className='rectangle' />
                    <h1 className='heading'>Good Morning, Johny!</h1>
                    <p className='subheading'>
                        üå§ It's better to go shopping before this friday
                    </p>
                    <div className='form'>
                        <div className='form-header'>
                            <TheFridgeTextInput
                                title='üçâ Item Name'
                                titleStyle={{
                                    color: "#2E3849",
                                    fontFamily: "Inter",
                                    fontWeight: 600,
                                    fontSize: 14,
                                    fontStyle: "normal",
                                }}
                                textInputContainerStyle={{
                                    width: 290,
                                    height: 40,
                                    borderRadius: 4,
                                    border: "2px solid #CCD6E1",
                                }}
                                type='text'
                                value={itemName}
                                onChangeText={(text: string) => setItemName(text)}
                            />
                            <TheFridgeTextInput
                                title='‚è∞ Expiry Date'
                                titleStyle={{
                                    color: "#2E3849",
                                    fontFamily: "Inter",
                                    fontWeight: 600,
                                    fontSize: 14,
                                    fontStyle: "normal",
                                }}
                                textInputContainerStyle={{
                                    width: 290,
                                    height: 40,
                                    borderRadius: 4,
                                    border: "2px solid #CCD6E1",
                                }}
                                containerStyle={{
                                    marginLeft: 8,
                                }}
                                type='text'
                                value={expiryDate}
                                onChangeText={(text: string) => setExpiryDate(text)}
                            />
                            <TheFridgeButton
                                buttonTitle='ADD TO FRIDGE'
                                buttonTitleStyle={{
                                    fontFamily: "Inter",
                                    fontWeight: "700",
                                    fontSize: 13,
                                    color: "#FFFFFF",
                                    textAlign: "center",
                                }}
                                containerStyle={{
                                    width: 230,
                                    height: 40,
                                    borderRadius: 4,
                                    backgroundColor: "#00598D",
                                    display: "flex",
                                    justifyContent: "center",
                                    alignItems: "center",
                                    marginLeft: 8,
                                    marginTop: 15,
                                }}
                                onItemClick={handleAddToFridge}
                            />
                        </div>
                        <div className='form-subText-container'>
                            <p className='form-subText'>
                                ‚ö†Ô∏è We don't want more than one piece of the same food in our
                                fridge.
                            </p>
                        </div>
                    </div>
                </div>
                {isLoading ? (
                    <div className='loding'>
                        Loading fridge items..
                    </div>
                ) : (
                    <div>
                        {apiData?.map((apiData, index) => {
                            return (
                                <div key={index}>
                                    <div className='text-wrapper'>Total items ‚Äî {dataLength}</div>
                                    <div className='item' >
                                        <div className='title'>
                                            {apiData.title}
                                        </div>
                                        <p className='expiry-date'>
                                            {apiData.expiry}
                                        </p>
                                        {expiryDateCheck(apiData.expiry) === 'Healthy' ?
                                            (
                                                <div className='chip'>
                                                    <div className='overlap-group-healthy'>
                                                        <div className='text text-healthy'>
                                                            Healthy
                                                        </div>
                                                    </div>
                                                </div>
                                            ) : (
                                                expiryDateCheck(apiData.expiry) === 'Expiring soon'
                                            ) ? (
                                                <div className='chip'>
                                                    <div className='overlap-group-expiring-soon'>
                                                        <div className='text text-expiring-soon'>
                                                            Expiring soon
                                                        </div>
                                                    </div>
                                                </div>
                                            ) : (
                                                <div className='chip'>
                                                    <div className='overlap-group-expired'>
                                                        <div className='text text-expired'>
                                                            Expired
                                                        </div>
                                                    </div>
                                                </div>
                                            )
                                        }
                                        <div className='delete-outline-black'>
                                            <DeleteIcon />
                                        </div>
                                    </div>
                                </div>
                            )
                        })}
                    </div>
                )}
            </div>
        </div>
    );
};

export default Dashboard;
