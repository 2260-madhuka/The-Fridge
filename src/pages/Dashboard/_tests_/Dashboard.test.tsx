import React from 'react';
import Dashboard from '../Dashboard';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';

jest.mock('axios');

const component = <Dashboard />;

describe('renders correctly', () => {
  jest.useFakeTimers();

  it('renders Snapshot', () => {
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });
});
