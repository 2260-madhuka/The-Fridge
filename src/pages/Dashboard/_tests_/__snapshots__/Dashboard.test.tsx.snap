// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly renders Snapshot 1`] = `
<div>
  <div
    class="screen"
  >
    <div
      class="div"
    >
      <div
        class="overlap"
      >
        <div
          class="rectangle"
        />
        <h1
          class="heading"
        >
          Good Morning, Johny!
        </h1>
        <p
          class="subheading"
        >
          üå§ It's better to go shopping before this friday
        </p>
        <div
          class="form"
        >
          <div
            class="form-header"
          >
            <div>
              <div>
                <div
                  style="border-width: 1px; border-color: #9b9b9b; border-radius: 5px; min-height: 48px; justify-content: center; flex-direction: row; align-items: center;"
                >
                  <div
                    style="margin-bottom: 4px;"
                  >
                    <p
                      style="color: rgb(46, 56, 73); font-family: Inter; font-weight: 600; font-size: 14px; font-style: normal;"
                    >
                      üçâ Item Name
                    </p>
                  </div>
                  <div>
                    <text>
                      <input
                        name="textInput"
                        style="width: 290px; height: 40px; border-radius: 4px; border: 2px solid #ccd6e1;"
                        type="text"
                        value=""
                      />
                    </text>
                  </div>
                </div>
              </div>
            </div>
            <div
              style="margin-left: 8px;"
            >
              <div
                style="margin-left: 8px;"
              >
                <div
                  style="border-width: 1px; border-color: #9b9b9b; border-radius: 5px; min-height: 48px; justify-content: center; flex-direction: row; align-items: center;"
                >
                  <div
                    style="margin-bottom: 4px;"
                  >
                    <p
                      style="color: rgb(46, 56, 73); font-family: Inter; font-weight: 600; font-size: 14px; font-style: normal;"
                    >
                      ‚è∞ Expiry Date
                    </p>
                  </div>
                  <div>
                    <text>
                      <input
                        name="textInput"
                        style="width: 290px; height: 40px; border-radius: 4px; border: 2px solid #ccd6e1;"
                        type="text"
                        value=""
                      />
                    </text>
                  </div>
                </div>
              </div>
            </div>
            <div
              style="width: 230px; height: 40px; border-radius: 4px; background-color: rgb(0, 89, 141); display: flex; justify-content: center; align-items: center; margin-left: 8px; margin-top: 15px;"
            >
              <text />
              <div>
                <div
                  style="display: flex; flex-direction: row;"
                >
                  <div
                    style="margin-right: 17px;"
                  >
                    <text
                      style="font-family: Inter; font-weight: 700; font-size: 13px; color: rgb(255, 255, 255); text-align: center;"
                    >
                      ADD TO FRIDGE
                    </text>
                  </div>
                  <div
                    style="display: flex; justify-content: center; align-items: center;"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="form-subText-container"
          >
            <p
              class="form-subText"
            >
              ‚ö†Ô∏è We don't want more than one piece of the same food in our fridge.
            </p>
          </div>
        </div>
      </div>
      <div
        class="loding"
      >
        Loading fridge items..
      </div>
    </div>
  </div>
</div>
`;
